byron/ledger/executable-spec/src/Ledger/Update.hs

```diff
@@ -1437,24 +1476,19 @@ instance STS PVBUMP where
     (ProtVer, PParams)
 
   type Signal PVBUMP = ()
-  data PredicateFailure PVBUMP
-    = NewEpoch
-    | OldEpoch
+
+  -- PVBUMP has no predicate failures
+  data PredicateFailure PVBUMP = NoPVBUMPFailure
     deriving (Eq, Show)
 
   initialRules = []
   transitionRules =
     [ do
         TRC ((s_n, fads, k), (pv, pps), ()) <- judgmentContext
-        let
-          mFirstStableSlot = minusSlotMaybe s_n (SlotCount . (2 *) . unBlockCount $ k)
-          r = case mFirstStableSlot of
-                Nothing -> []
-                Just s  -> filter ((<= s) . fst) fads
-        if r == []
-          then pure $! (pv, pps)
-          else do
-            let (_, (pv_c, pps_c)) = last r
```

Comment:
"The last shall be first"

![image](https://user-images.githubusercontent.com/175315/62205251-55a3bc80-b38f-11e9-8431-26e8a2069397.png)

Reply:
Amen